server/fleet/apple_mdm.go

```diff
@@ -444,6 +444,17 @@ func (p MDMAppleSetupPayload) AuthzType() string {
 	return "mdm_apple_settings" // TODO: add mdm_apple_setup to rego?
 }
 
+// HostDEPAssignment represents a row in the host_dep_assignments table
+type HostDEPAssignment struct {
+	HostID    uint       `db:"host_id"`
+	AddedAt   time.Time  `db:"added_at"`
+	DeletedAt *time.Time `db:"deleted_at"`
+}
+
+func (h HostDEPAssignment) IsDEPAssignedToFleet() bool {
```

Comment:
If you make that a method on `*HostDEPAssignment`, you can check `if h == nil { return false }` before the condition and then the caller can call the method even if the `*HostDEPAssignment` it has is `nil` (cleans up the callsite a bit).